LEGO1/mxdisplaysurface.cpp

```diff
@@ -216,9 +248,63 @@ void MxDisplaySurface::Clear()
 	this->Reset();
 }
 
-// STUB: LEGO1 0x100baae0
+// FUNCTION: LEGO1 0x100baae0
 void MxDisplaySurface::SetPalette(MxPalette* p_palette)
 {
+	HDC hdc;
```

Comment:
Something must be missing here at the start, because the original grabs `0x414` from the stack and we're using about `0x400` less than that. They could be unused variables. Maybe two MxU16 arrays of 256 colors?

Reply:
Actually this might be another case like the `BITMAPINFO` wrapper in `MxBitmap` where they have defined a wrapper around `LOGPALETTE` where the `palPalEntry` array has 256 entries instead of 1. That would put the size of `lpal` at `0x404` or 1028 bytes